internal/controllers/operator_controller_test.go

```diff
@@ -1018,6 +1023,94 @@ var _ = Describe("Operator Controller Test", func() {
 			Expect(cond.Message).To(Equal("installation has not been attempted as spec is invalid"))
 		})
 	})
+	When("a catalog changes on cluster", func() {
```

Comment:
We may want to consider sectioning these tests off in a separate test file, or at least root level `Describe` block. Reasoning is that everything else here is assuming no manager, no controller, and instead manually running `Reconcile` with pre-setup and post-assertions.

These tests start a manager, use `Eventually`, etc.